tests/test_slim_engine.py

```diff
@@ -373,6 +373,52 @@ def count_mut_types(ts):
         n_mut_types = count_mut_types(ts)
         assert n_mut_types[0] > 0
         assert n_mut_types[1] > 0
+        # test DFE with dfe-annotation
+        with tempfile.NamedTemporaryFile(mode="w") as f:
+            cmd = (
+                f"-q -e slim --slim-scaling-factor 20 --slim-path {slim_path} "
+                f"HomSap -c chr22 -o {f.name} "
+                "-d OutOfAfrica_3G09 --dfe Gamma_K17 "
+                "--dfe-annotation ensembl_havana_104_CDS 10"
+            ).split()
+            capture_output(stdpopsim.cli.stdpopsim_main, cmd)
+            ts = tskit.load(f.name)
+        assert ts.num_samples == 10
+        assert all(tree.num_roots == 1 for tree in ts.trees())
+        n_mut_types = count_mut_types(ts)
+        assert n_mut_types[0] > 0
+        assert n_mut_types[1] > 0
+        # test DFE with interval and no dfe
+        with tempfile.NamedTemporaryFile(mode="w") as f:
+            cmd = (
+                f"-q -e slim --slim-scaling-factor 20 --slim-path {slim_path} "
+                f"HomSap -c chr22 -l0.01 -o {f.name} "
+                "-d OutOfAfrica_3G09 "
+                "--dfe-interval 1000,100000 10"
+            ).split()
+            with pytest.raises(SystemExit):
```

Comment:
If you do
```suggestion
           with pytest.raises(SystemExit, match="interval has been assigned without a DFE"):
```
then that'll test that the right error is called.

Reply:
nice